src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -698,4 +642,158 @@ public static void constructLobs(Random random, IonType type, IonWriter writer)
+        File file = new File(path);
+        try (IonWriter writer = WriteRandomIonValues.formatWriter(format, file)) {
+            WriteRandomIonValues.writeRequestedSizeFile(size, writer, file, IonType.STRUCT,fields, annotations);
+            WriteRandomIonValues.printInfo(path);
+        }
+    }
+
+    /**
+     * Construct Ion structs based on the provided constraints.
+     * @param fields is an Ion struct which contains the constraints needed during the data generating process.
+     * @param annotations is an Ion list which contains the required annotations.
+     * @param writer writes Ion struct data.
+     * @throws IOException if errors occur when writing data.
+     */
+    public static void constructIonStruct(IonStruct fields, IonList annotations, IonWriter writer) throws IOException {
+        Random random = new Random();
+        try (IonReader reader = IonReaderBuilder.standard().build(fields)) {
+            reader.next();
+            reader.stepIn();
+            int index = random.nextInt(annotations.size());
+            writer.addTypeAnnotation(String.valueOf(annotations.get(index)));
+            writer.stepIn(IonType.STRUCT);
+            while (reader.next() != null) {
+                IonSystem system = IonSystemBuilder.standard().build();
+                String fieldName = reader.getFieldName();
+                IonValue struct = system.newValue(reader);
+                IonStruct value = (IonStruct) struct;
+
+                // if the value of "occurs" is optional, the integer represents this value is 1 or 0.
+                int occurTime = WriteRandomIonValues.parseConstraints(value, "occurs");
+                if (occurTime == 0) {
+                    continue;
+                }
+                writer.setFieldName(fieldName);
+                IonType type = null;
+                type = IonType.valueOf(value.get("type").toString().toUpperCase());
+                switch (type) {
+                    // if more types of Ion data are available, the logic should be added below.
+                    case STRING:
+                        String codepointRange = "[0, 1114111]";
+                        List<Integer> range = parseRange(codepointRange);
+                        int codePointsLengthBound = 20;
+                        if (value.get("codepoint_length") != null) {
+                            codePointsLengthBound = WriteRandomIonValues.parseConstraints(value, "codepoint_length");
+                        }
+                        String stringValue = WriteRandomIonValues.constructString(range, codePointsLengthBound);
+                        writer.writeString(stringValue);
+                        break;
+                    case TIMESTAMP:
+                        // call the function that extract the precision of the timestamp
+                        Timestamp.Precision[] precisions = Timestamp.Precision.values();
```

Comment:
`Timestamp.Precision.values()` will allocate a new array each time it is called. Since it never changes, consider storing its result as a constant.

Reply:
> `Timestamp.Precision.values()` will allocate a new array each time it is called. Since it never changes, consider storing its result as a constant.

Set all precisions as a constant, changes have been updated.